client/landing/stepper/declarative-flow/internals/steps-repository/importer/index.tsx

```diff
@@ -77,10 +82,26 @@ export function withImporterWrapper( Importer: ImporterCompType ) {
 			migrationStatus === MigrationStatus.BACKING_UP ||
 			migrationStatus === MigrationStatus.BACKING_UP_QUEUED ||
 			migrationStatus === MigrationStatus.RESTORING;
+		const userId = useSelector( getCurrentUserId );
+		const isEmailVerified = useSelector( isCurrentUserEmailVerified );
+		const currentPath = window.location.pathname + window.location.search;
+		const pathStep = isEmailVerified ? null : currentPath;
+		/**
+		 * Save the selected trial plan path in the user's preferences.
+		 */
+		const saveHostingFlowPathStep = () => {
+			dispatch( savePreference( `hosting-flow-path-step-${ userId }`, pathStep ) );
+		};
 
 		/**
 	 	↓ Effects
 		 */
+		useEffect( () => {
+			if ( flow === IMPORT_HOSTED_SITE_FLOW ) {
+				saveHostingFlowPathStep();
+			}
```

Comment:
Would it make sense to not call this at all if `isEmailVerified` is true?

Reply:
Probably. My thinking here was that it would reset/delete the calypso preference if the email is verified but perhaps that should be done explicitly elsewhere.

Reply:
I've added a check on `isEmailVerified` in the new hook